server/schema/squiggle/squiggle.resolver.js

```diff
@@ -33,6 +33,34 @@ module.exports = {
       throw APIError(null, error);
     }
   },
+
+  // added code to update squiggle here..
+  updateSquiggle: async (
+    _parent,
+    { id, newSquiggleType, newContent },
+    { session, authToken, decodedToken, API: { Squiggle } }
+  ) => {
+    try {
+      if (!UserPermission.exists(session, authToken, decodedToken, 'squiggle.write.update')) {
+        throw APIError('FORBIDDEN', null, {
+          reason: 'The user does not have the required permissions to update squiggles.',
+        });
+      }
+      const _squiggle = await Squiggle.findByID(id);
+
+      if (!_squiggle) {
+        throw APIError('NOT_FOUND', null, { reason: 'The squiggle to update does not exist.',});
+      }
+      _squiggle.squiggleType = newSquiggleType;
+      _squiggle.content = newContent;
+  
+      await _squiggle.save();
```

Comment:
@Srish-ty do all the mongoose related work in squiggle datasource file.

Reply:
done. [commit](https://github.com/Monday-Morning/project-reclamation/pull/205/commits/82bb74d89cb609214b7a3db08adf348703b34e7b)